PyRDF/backend/Backend.py

```diff
@@ -93,7 +97,7 @@ def check_supported(self, operation_name):
             )
 
     @abstractmethod
-    def execute(self, generator):
+    def execute(self, generator, trigger_loop=False):
```

Comment:
I like this, now that I see it I wish it were added before. One counterargument to it though: in this way we are creating pyroot nodes even when no computation is involved, or not ?

I am thinking a situation in which the user first wants to find out the name of a column
```
df = PyRDF.RDataFrame("some_tree","some_file.root")

colnames = df.GetColumnNames() # this will create pyroot nodes and return a list of strings
```
and then executes some operations
```
filter1 = df.Filter("valid c++ filter")
histo1 = filter1.Histo1D("col1")

histo1.Draw() # this triggers the event loop, recreating the same pyroot objects
```

A way to solve this would be to implement some checks in the `mapper` function of the `CallableGenerator`, e.g. 
```
if not node_py.pyroot_node:
    RDFOperation...
    [...]
```